api/v1/helpers/verbs/doGet.js

```diff
@@ -29,11 +30,33 @@ const httpStatus = require('../../constants').httpStatus;
  *  resource type to retrieve.
  */
 function doGet(req, res, next, props) {
-  u.findByKey(props, req.swagger.params)
-  .then((o) => {
-    res.status(httpStatus.OK).json(u.responsify(o, props, req.method));
-  })
-  .catch((err) => u.handleError(next, err, props.modelName));
+  if (props.cacheEnabled) {
+    redisCache.get(req.swagger.params.key.value, (cacheErr, reply) => {
```

Comment:
Let's say perspective "XYZ" was cached based on a call to `GET /v1/perspectives/XYZ?fields=name`. Now when I call `GET /v1/perspectives/XYZ` I'm going to get the cached response, which means the perspective I get back only contains the "name" attribute, but `GET /v1/perspectives/XYZ` without any `?fields=...` is supposed to return *all* the perspective fields...